src/psyclone/psyir/backend/fortran.py

```diff
@@ -1055,16 +1055,18 @@ def filecontainer_node(self, node):
         :rtype: str
 
         :raises VisitorError: if the attached symbol table contains \
-            any data symbols.
+            any non-routine symbols.
         :raises VisitorError: if more than one child is a Routine Node \
             with is_program set to True.
 
         '''
         if node.symbol_table.symbols:
-            raise VisitorError(
-                f"In the Fortran backend, a file container should not have "
-                f"any symbols associated with it, but found "
-                f"{len(node.symbol_table.symbols)}.")
+            for symbol in node.symbol_table.symbols:
+                if not isinstance(symbol, RoutineSymbol):
```

Comment:
@sergisiso this currently doesn't create a container symbol as you expected. My concern from reading the container symbol docs is that the container symbol feels more like it's representing a 'use' statement than anything else.
It's possible we could create a new module interface to use for this type of module declaration (instead of FortranModuleInterface) but I'm still unclear that if `mysub` used the declared module here that PSyclone would work correctly

Reply:
I'm a bit confused here as #1298 is closed however there is a TODO in Container symbol for 1298 @sergisiso

Reply:
Ok, I will have a look, but in any case it seems this needs a bit of work, so no need to do it in this PR.

Reply:
Ok - I changed title to "towards" 2201 as we still have some remaining issues.